tests/test_exposure.py

```diff
@@ -777,6 +777,88 @@ def testGetCutout(self):
                     with self.assertRaises(pexExcept.InvalidParameterError, msg=msg):
                         self.smallExposure.getCutout(cutoutCenter, cutoutSize)
 
+    def testGetConvexPolygon(self):
+        """Test the convex polygon."""
+        # Check that we do not have a convex polygon for the plain exposure.
+        self.assertIsNone(self.exposureMiOnly.convex_polygon)
+
+        # Check that all the points in the bounding box are in the polygon
+        bbox = self.exposureMiWcs.getBBox()
+        # Grow by the default padding.
+        bbox.grow(10)
+        x, y = np.meshgrid(np.arange(bbox.getBeginX(), bbox.getEndX()),
+                           np.arange(bbox.getBeginY(), bbox.getEndY()))
+        wcs = self.exposureMiWcs.wcs
+        ra, dec = wcs.pixelToSkyArray(x.ravel().astype(np.float64),
+                                      y.ravel().astype(np.float64))
+
+        poly = self.exposureMiWcs.convex_polygon
+        contains = poly.contains(ra, dec)
+        np.testing.assert_array_equal(contains, np.ones(len(contains), dtype=bool))
+
+        # Check that points outside of the bounding box are not in the polygon
+        bbox.grow(1)
+
+        ra, dec = wcs.pixelToSkyArray(
+            np.linspace(bbox.getBeginX(), bbox.getEndX(), 100),
+            np.full(100, bbox.getBeginY()))
+        contains = poly.contains(ra, dec)
+        np.testing.assert_array_equal(contains, np.zeros(len(contains), dtype=bool))
+
+        ra, dec = wcs.pixelToSkyArray(
+            np.linspace(bbox.getBeginX(), bbox.getEndX(), 100),
+            np.full(100, bbox.getEndY()))
+        contains = poly.contains(ra, dec)
+        np.testing.assert_array_equal(contains, np.zeros(len(contains), dtype=bool))
+
+        ra, dec = wcs.pixelToSkyArray(
+            np.full(100, bbox.getBeginX()),
+            np.linspace(bbox.getBeginY(), bbox.getEndY(), 100))
+        contains = poly.contains(ra, dec)
+        np.testing.assert_array_equal(contains, np.zeros(len(contains), dtype=bool))
+
+        ra, dec = wcs.pixelToSkyArray(
+            np.full(100, bbox.getEndX()),
+            np.linspace(bbox.getBeginY(), bbox.getEndY(), 100))
+        contains = poly.contains(ra, dec)
+        np.testing.assert_array_equal(contains, np.zeros(len(contains), dtype=bool))
+
+    def testContainsSkyCoords(self):
+        """Test the sky coord containment code."""
+        self.assertRaises(ValueError,
```

Comment:
Please use `RaisesRegex` here, so we can be sure that the exception is coming from where we think it is. (I've just had to deal with some tests that weren't getting the exception the author thought, even though it had the right type! So this is on my mind.)